File: 1849-maximum-absolute-sum-of-any-subarray/maximum-absolute-sum-of-any-subarray.py
class Solution:
    def maxAbsoluteSum(self, nums: List[int]) -> int:
        sum, minSum, maxSum = 0, 0, 0
        for num in nums:
            sum += num
            maxSum = max(sum, maxSum)
            minSum = min(sum, minSum)
        return abs(maxSum - minSum)

File: 1849-maximum-absolute-sum-of-any-subarray/maximum-absolute-sum-of-any-subarray.c
int maxAbsoluteSum(int* nums, int numsSize) {
    int sum = 0, maxsum = 0, minsum = 0;
    for (int i=0; i<numsSize; i++){
        sum += nums[i];
        if (maxsum < sum) maxsum = sum;
        if (minsum > sum) minsum = sum;
    }
    return abs(maxsum - minsum);
}